bob-apps/bin/bobc.rs

```diff
@@ -1,137 +1,421 @@
+        Ok(data) => {
+            let timestamp = SystemTime::now()
+                .duration_since(UNIX_EPOCH)
+                .unwrap()
+                .as_secs();
+            let meta = BlobMeta { timestamp };
+            let blob = Blob {
+                data,
+                meta: Some(meta),
+            };
+            let message = PutRequest {
+                key: Some(BlobKey { key: get_key_value(key, key_size) }),
+                data: Some(blob),
+                options: None,
+            };
+            let put_req = Request::new(message);
+
+            match client.put(put_req).await {
+                Ok(_) => info!("key: {}, file: {}", key, filename),
+                Err(e) => error!("key: {}, file: {}, error: {:?}", key, filename, e)
+            }
+        }
+        Err(e) => {
+            error!("key: {}, file: {}, error: {:?}", key, filename, e)
+        }
+    }
+}
 
+async fn get(key: u64, key_size: usize, filename: &str, client: &mut BobApiClient<Channel>) {
     let message = GetRequest {
-        key: Some(BlobKey { key }),
+        key: Some(BlobKey { key: get_key_value(key, key_size) }),
         options: None,
     };
     let get_req = Request::new(message);
     let res = client.get(get_req).await;
     match res {
         Ok(res) => {
             let res: tonic::Response<_> = res;
-            let data: &Blob = res.get_ref();
-            info!("OK");
-            info!("response meta: {:?})", res.metadata());
-            info!("data len: {}, meta: {:?}", data.data.len(), data.meta);
+            let data = res.get_ref();
+            match fs::write(filename, &data.data).await {
+                Err(e) => error!("key: {}, file: {}, error: {:?}", key, filename, e),
+                _ => info!("key: {}, file: {}", key, filename)
+
+            }
         }
-        Err(res) => {
-            info!("ERR");
-            info!("{:?}", res);
+        Err(e) => {
+            error!("key: {}, file: {}, error: {:?}", key, filename, e)
+        }
+    }
+}
+
+async fn exist(keys: Vec<u64>, key_size: usize, client: &mut BobApiClient<Channel>) {
+    let k = keys.iter().map(|key| BlobKey { key: get_key_value(*key, key_size) }).collect();
+    let message = ExistRequest {
+        keys: k,
+        options: None,
+    };
+    let request = Request::new(message);
+    let res = client.exist(request).await;
+    match res {
+        Ok(res) => {
+            let res: tonic::Response<_> = res;
+            let data = res.get_ref();
+            for (k, r) in iter::zip(keys, &data.exist) {
+                info!("key: {}, exists: {}", k, r)
+            }
+        }
+        Err(e) => {
+            error!("{:?}", e);
         }
     }
 }
 
 fn get_matches<'a>() -> ArgMatches<'a> {
-    let key_arg = Arg::with_name("key").takes_value(true).required(true);
-    let uri_arg = Arg::with_name("uri")
+    let key_arg = Arg::with_name(KEY_ARG)
+        .short("k")
+        .long(KEY_ARG)
```

Comment:
It is better to use hardcoded string for `long` names, because this improves readability and allow to modify long name without changing internal argument name

Reply:
Done: [ca6aaee](https://github.com/qoollo/bob/pull/562/commits/ca6aaee23fab4471a2fa9ddeab2e5ed4d28c9719)